### PR TITLE
Otoken: Burn and Mint

### PR DESCRIPTION
**Pull request by antoncoding**
### Description

This PR add `mintOtoken` and `burnOtoken` functions   
This should be reviewed after #24 is merged, and change the based branch to master.

### Todos:
- [x] Add doc for each function
- [x] Make test caes 100% coverage

### CODE REVIEW

File: contracts/Otoken.sol [hunk 1]
```diff
@@ -1,12 +1,242 @@
...
+        bool _isPut
+    ) external initializer {
+        underlyingAsset = _underlyingAsset;
+        strikeAsset = _strikeAsset;
+        collateralAsset = _collateralAsset;
+        strikePrice = _strikePrice;
+        expiry = _expiry;
+        isPut = _isPut;
+        (string memory name, string memory symbol) = _getNameAndSymbol();
+        __ERC20_init_unchained(name, symbol);
```
Comment (aparnakr) [inline contracts/Otoken.sol:70]: where are the decimals set?

File: contracts/Otoken.sol [hunk 2]
```diff
@@ -1,12 +1,242 @@
...
+
+    /**
+     * @notice Mint oToken for an account.
+     * @dev this is a Controller only method. Access control is taken care of by _beforeTokenTransfer hook.
+     * @param account the account to mint token to
+     * @param amount the amount to mint
+     */
+    function mintOtoken(address account, uint256 amount) external {
+        _mint(account, amount);
+    }
```
Comment (aparnakr) [inline contracts/Otoken.sol:81]: isn't this onlyController?

File: contracts/Otoken.sol [hunk 3]
```diff
@@ -1,12 +1,242 @@
...
+        if (number > 99) return Strings.toString(number % 100);
+        if (number < 10) {
+            return string(abi.encodePacked("0", str));
+        } else {
+            return str;
+        }
+    }
+
+    /**
+     * @dev return string of optino type
```
Comment (aparnakr) [inline contracts/Otoken.sol:172]:
```suggestion
     * @dev return string of option type
```

File: contracts/Otoken.sol [hunk 4]
```diff
@@ -1,12 +1,242 @@
...
+            return ("NOV", "November");
+        } else {
+            return ("DEC", "December");
+        }
+    }
 
+    /**
+     * @dev this function overrides the _beforeTokenTransfer hook in ERC20Initializable.sol.
+     * If the operation is mint or burn, requires msg.sender to be the controller.
+     * @dev the functino signature is the same as _beforeTokenTransfer defined in ERC20Initializable.sol.
```
Comment (aparnakr) [inline contracts/Otoken.sol:220]:
```suggestion
     * the function signature is the same as _beforeTokenTransfer defined in ERC20Initializable.sol.
```

File: contracts/Otoken.sol (hunk 1 above)
Comment (antoncoding) [reply contracts/Otoken.sol:70]: is set in the `__ERC20_init_unchained ` function

```javascript
function __ERC20_init_unchained(string memory name, string memory symbol) internal initializer {
        _name = name;
        _symbol = symbol;
        _decimals = 18;
    }
```

**Pull request merged**